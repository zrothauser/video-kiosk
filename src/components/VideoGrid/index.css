.b-video-grid {
  display: flex;
  flex-flow: row wrap;
  width: 100%;
}
