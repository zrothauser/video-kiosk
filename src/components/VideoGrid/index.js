// Dependencies
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const VideoGrid = props => (
  <div>
    {props.videos.map(video => (
      <div key={video.id} style={{ borderBottom: '1px solid black', marginBottom: '1em' }}>
        <Link to={`/video/${video.id}`}>
          {video.title}
        </Link>
        <br />
        {video.id}<br />
        {video.description}<br />
        {video.vimeoURL}<br />
        {video.thumbnailLarge}<br />
      </div>
    ))}
  </div>
);

VideoGrid.propTypes = {
  videos: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default VideoGrid;
