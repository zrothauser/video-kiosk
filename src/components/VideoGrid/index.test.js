import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import VideoGrid from './index';

describe('components', () => {
  describe('VideoGrid', () => {
    const mockProps = {
      categorySlug: 'clyfford-still-basics',
      videos: [],
    };

    it('renders without crashing', () => {
      shallow(<VideoGrid {...mockProps} />);
    });

    it('renders correctly', () => {
      const VideoGridComponent = renderer.create(
        <Router>
          <VideoGrid {...mockProps} />
        </Router>,
      );

      expect(VideoGridComponent).toMatchSnapshot();
    });
  });
});
