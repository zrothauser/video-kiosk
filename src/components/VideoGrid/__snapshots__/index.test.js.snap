// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components VideoGrid renders correctly 1`] = `
<div
  className="b-video-grid"
/>
`;
